mod/tigron/internal/assertive/assertive.go

```diff
@@ -0,0 +1,169 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package assertive
+
+import (
+	"errors"
+	"strings"
+	"time"
+)
+
+type testingT interface {
+	Helper()
+	FailNow()
+	Fail()
+	Log(args ...interface{})
+}
+
+// ErrorIsNil immediately fails a test if err is not nil.
+func ErrorIsNil(t testingT, err error, msg ...string) {
+	t.Helper()
+
+	if err != nil {
+		t.Log("expecting nil error, but got:", err)
+		failNow(t, msg...)
+	}
+}
+
+// ErrorIs immediately fails a test if err is not the comparison error.
+func ErrorIs(t testingT, err, compErr error, msg ...string) {
+	t.Helper()
+
+	if !errors.Is(err, compErr) {
+		t.Log("expected error to be:", compErr, "- instead it is:", err)
+		failNow(t, msg...)
+	}
+}
+
+// IsEqual immediately fails a test if the two interfaces are not equal.
+func IsEqual(t testingT, actual, expected interface{}, msg ...string) {
```

Comment:
> ```suggestion
> func IsEqual(t testingT, expected, actual interface{}, msg ...string) {
> ```
> 
> To follow the existing standard:
> - https://pkg.go.dev/github.com/stretchr/testify/assert#Equal
> - https://junit.org/junit4/javadoc/4.8/org/junit/Assert.html
> - etc.
> 

_Originally posted by @AkihiroSuda in https://github.com/containerd/nerdctl/pull/4040#discussion_r2015745287_

Reply:
The problem I have with this is that it is not consistent internally.

testify ErrorIs for example is the other way round (first actual error, then target error), which I find confusing everytime.

Also, gotesttool is `actual`, `expected` as well:
https://github.com/gotestyourself/gotest.tools/blob/main/assert/assert.go#L188

Anyhow, I do not intend on making `assertive` public. People can and should use established assert libraries when they write their tests if they need them. This here is meant to stay internal.

Reply:
Going to mark this as resolved.
Suggestion is that _if_ in the future `assertive` moves from internal to public, we can, and should, have a discussion on API (and can have that conversation on parameters order).

Lmk if you still think this is an issue right now and I'll un-resolve.